framework/util/image_writer.cpp

```diff
@@ -47,14 +47,14 @@
 // zlib compress2 function resulted in a 40+% reduction versus the original image.
 uint8_t* GFXRECON_zlib_compress2(uint8_t* data, int32_t data_len, int32_t* out_len, int32_t quality)
 {
-    unsigned long alloc_len = compressBound(data_len);
-    uint8_t*      target    = reinterpret_cast<uint8_t*>(malloc(alloc_len));
+    unsigned long alloc_len = compressBound(static_cast<uLong>(data_len));
+    auto*         target    = reinterpret_cast<uint8_t*>(malloc(alloc_len));
```

Comment:
why change to auto?

Reply:
that is frequently advised by clang-tidy / my IDE:
"Clang-Tidy: Use auto when initializing with a cast to avoid duplicating the type name"

https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-auto.html

Reply:
and for pointer-types, best-practice advises to prefer `auto*` over `auto`